arithmetization/src/test/java/net/consensys/linea/zktracer/testing/BytecodeCompiler.java

```diff
@@ -46,6 +49,41 @@ private static Bytes toBytes(final int x) {
     return Bytes.ofUnsignedLong(x).trimLeadingZeros();
   }
 
+  /**
+   * Assemble an EVM program into bytecode. Requirement are: one instruction per line; `;` mark a
+   * line as a comment; no more than one word per line, safe for PUSHs.
+   *
+   * @param program
+   * @return the bytecode corresponding to the assembly code
+   */
+  public BytecodeCompiler assemble(final String program) {
+    BufferedReader bufReader = new BufferedReader(new StringReader(program));
+    String line;
+    while (true) {
+      try {
+        if ((line = bufReader.readLine()) == null) break;
+        if (line.isBlank()) continue;
+        if (line.startsWith(";")) continue;
+
+        String[] ls = line.split("\\s+");
+        final OpCode opCode = OpCode.fromMnemonic(ls[0]);
+        this.op(opCode);
+        final int pushSize = opCode.byteValue() - (int) OpCode.PUSH1.byteValue() + 1;
+        final boolean isPush = pushSize >= 1 && pushSize <= 32;
+        if (isPush) {
+          this.immediate(Bytes.fromHexString(ls[1], pushSize));
```

Comment:
Does this mean that we can write arbitrary size hexstrings after a PUSHX and it will only fail if there are fewer bytes than `pushSize` ?

Reply:
The opposite; if there are fewer bytes that `pushSize`, then `fromHexString` will left-pad them with 0's; if there are too many, it will throw an exception.